framework/helpers/BaseUrl.php

```diff
@@ -378,8 +378,7 @@ public static function home($scheme = false)
      */
     public static function isRelative($url)
     {
-        $urlComponents = parse_url($url, PHP_URL_SCHEME);
-        return strncmp($url, '//', 2) && empty($urlComponents);
+        return preg_match('~^[[:alpha:]][[:alnum:]+-.]*://|^//~', $url) === 0;
```

Comment:
I agree it wouldn't be problematic, but why make it less correct? What's the benefit?

Reply:
Hm, not sure why but in my head this feels faster? 🤣 strange how the mind works, hehe. Ok, @rob006 could you take a look?

Reply:
I think we should choose pattern that is faster - IMO it is not worth to have more strict check here to cover edge cases like 123://example.com, if it will be slower for basically all real-world use cases.

Also, IMO it should be `'~^[[:alpha:]][[:alnum:]+-.]+://|^//~'`, to not accept URLs like `://example.com`

Reply:
Both your examples are already not accepted. The first character has to be alpha, then 0 or more alphanumerics (and a few special chars).

Have you got data to show that this regex is slow? It is simple and has an anchor at the start, I don't see why it would be slow. I'm skeptical any change would be speeding it up immensely, since most time is spent in compilation not evaluation of the expression.